File: app/Models/SavingsGoal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class SavingsGoal extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'account_id',
        'user_id',
        'category_id',
        'subcategory_id',
        'name',
        'target_amount',
        'initial_amount',
        'currency',
        'tracking_mode',
        'start_date',
        'target_date',
        'settings',
    ];

    protected $casts = [
        'target_amount' => 'decimal:4',
        'initial_amount' => 'decimal:4',
        'start_date' => 'date',
        'target_date' => 'date',
        'settings' => 'array',
    ];

    public function account(): BelongsTo
    {
        return $this->belongsTo(Account::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function category(): BelongsTo
    {
        return $this->belongsTo(Category::class);
    }

    public function subcategory(): BelongsTo
    {
        return $this->belongsTo(Subcategory::class);
    }
}

File: ios/PersonalFinance/Models/SavingsGoal.swift
import Foundation

struct SavingsGoal: Decodable, Identifiable, Hashable {
    let id: Int
    let name: String
    let targetAmount: MoneyValue
    let initialAmount: MoneyValue?
    let currency: String
    let trackingMode: String?
    let startDate: String?
    let targetDate: String?
    let progress: SavingsGoalProgress?
}

struct SavingsGoalProgress: Decodable, Hashable {
    let currentAmount: MoneyValue
    let contributed: MoneyValue
    let remaining: MoneyValue
    let percentage: Double
    let isComplete: Bool
}
